src/kubeobjects/istio.go

```diff
@@ -0,0 +1,237 @@
+		Group:   IstioGVRName,
+		Version: "v1alpha3",
+		Kind:    "VirtualService",
+	}
+
+	// ServiceEntryGVK => definition of virtual service GVK for oneagent
+	ServiceEntryGVK = schema.GroupVersionKind{
+		Group:   IstioGVRName,
+		Version: "v1alpha3",
+		Kind:    "ServiceEntry",
+	}
+)
+
+// CheckIstioEnabled checks if Istio is installed
+func CheckIstioEnabled(cfg *rest.Config) (bool, error) {
+	discoveryclient, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return false, err
+	}
+	apiGroupList, err := discoveryclient.ServerGroups()
+	if err != nil {
+		return false, err
+	}
+
+	for _, apiGroup := range apiGroupList.Groups {
+		if apiGroup.Name == IstioGVRName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// BuildNameForEndpoint returns a name to be used as a base to identify Istio objects.
+func BuildNameForEndpoint(name string, protocol string, host string, port uint32) string {
+	sum := sha256.Sum256([]byte(fmt.Sprintf("%s-%s-%s-%d", name, protocol, host, port)))
+	return hex.EncodeToString(sum[:])
+}
+
+func buildObjectMeta(name string, namespace string) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      name,
+		Namespace: namespace,
+	}
+}
+
+func BuildIstioLabels(name, role string) map[string]string {
+	return map[string]string{
+		"dynatrace":            "oneagent",
+		"oneagent":             name,
+		"dynatrace-istio-role": role,
+	}
+}
+
+func isIp(host string) bool {
+	return net.ParseIP(host) != nil
+}
+
+func VerifyIstioCrdAvailability(instance *dynatracev1beta1.DynaKube, config *rest.Config) probeResult {
+	var probe probeResult
+
+	probe, _ = kubernetesObjectProbe(ServiceEntryGVK, instance.GetNamespace(), "", config)
+	if probe == ProbeTypeNotFound {
+		return probe
+	}
+
+	probe, _ = kubernetesObjectProbe(VirtualServiceGVK, instance.GetNamespace(), "", config)
+	if probe == ProbeTypeNotFound {
+		return probe
+	}
+
+	return ProbeTypeFound
+}
+
+func kubernetesObjectProbe(gvk schema.GroupVersionKind,
+	namespace string, name string, config *rest.Config) (probeResult, error) {
+
+	var objQuery unstructured.Unstructured
+	objQuery.Object = make(map[string]interface{})
+
+	objQuery.SetGroupVersionKind(gvk)
+
+	runtimeClient, err := client.New(config, client.Options{})
+	if err != nil {
+		return probeUnknown, err
+	}
+	if name == "" {
+		err = runtimeClient.List(context.TODO(), &objQuery, client.InNamespace(namespace))
+	} else {
+		err = runtimeClient.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: name}, &objQuery)
+	}
+
+	return mapErrorToObjectProbeResult(err)
+}
+
+func mapErrorToObjectProbeResult(err error) (probeResult, error) {
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return probeObjectNotFound, err
+		} else if meta.IsNoMatchError(err) {
+			return ProbeTypeNotFound, err
+		}
+
+		return probeUnknown, err
+	}
+
+	return probeObjectFound, nil
+}
+
+func HandleIstioConfigurationForServiceEntry(instance *dynatracev1beta1.DynaKube,
```

Comment:
`serviceentry.go` under the `istio` package
you shouldn't pass in a logger, use the package level logger instead